guides/hack/02-source-code-fundamentals/19-literals.md

```diff
@@ -86,9 +86,52 @@ Escape sequence | Character name | Unicode character
 \xhh or \Xhh  | 1-2-digit hexadecimal digit value hh | U+00hh
 \u{xxxxxx} | UTF-8 encoding of Unicode codepoint U+xxxxxx | U+xxxxxx
 
-Within a double-quoted string literal a dollar ($) character *not* escaped by a backslash (\\) is handled using *variable
-substitution rules*, which follow.
+Within a double-quoted string literal a dollar ($) character *not* escaped by a backslash (\\) is handled using *[variable
+substitution rules](#variable-substitution)*.
 
+### Heredoc String Literals
+
+A heredoc string literal is a string literal delimited by `<<< id` and `id`. The literal can contain any source character.
+Certain other (and sometimes non-printable) characters can also be expressed as [escape sequences](#string-literals__double-quoted-string-literals).
+
+For example:
+
+```
```

Comment:
IIRC if you add a filename here (```heredoc.hack) it will wrap the code in an `__EntryPoint` function automatically, but *only in the extracted file* -- the guide will only show the exact part of the code that you have here.

This could be the best compromise -- we'd still have a "real" example that is executed and typechecked, but without having any boilerplate code cluttering the guide text.

Reply:
@jjergus I actually did want to create an automated test for both of these but it looks like they flag incorrectly as errors (I think!) when building. Based on some internal threads multi-line needs to be wrapped in <? ?> to execute correctly VIA cli?

Reply:
added variable examples